src/cbmc_starter_kit/update.py

```diff
@@ -0,0 +1,259 @@
+    options = [
+        {'flag': '--cbmc-root',
+         'type': Path,
+         'metavar': 'CBMC',
+         'help': 'Root of CBMC proof infrastructure (default: ".")'},
+        {'flag': '--starter-kit-root',
+         'type': Path,
+         'metavar': 'STARTER_KIT',
+         'help': """
+          Root of CBMC starter kit submodule (default: None or root of starter kit submodule
+          installed in repository containing CBMC)"""},
+        {'flag': '--no-migrate',
+         'action': 'store_true',
+         'help': """
+          Do not remove symlinks under CBMC.
+          Normally remove symlinks under CBMC to files under STARTER_KIT."""},
+        {'flag': '--no-test-removal',
+         'action': 'store_true',
+         'help': """
+          Do not remove negative tests in CBMC/negative_tests.
+          Normally remove the directory CBMC/negative_tests since most projects don't
+          use these tests."""},
+        {'flag': '--no-update',
+         'action': 'store_true',
+         'help': """
+          Do not update Makefile.common and run-cbmc-proofs.py under CBMC/proofs.
+          Normally update these files with the versions in the starter kit package."""},
+        {'flag': '--remove-starter-kit-submodule',
+         'action': 'store_true',
+         'help': """
+          Remove the starter kit submodule if it is present.
+          Normally just recommend removal."""},
+        {'flag': '--remove-litani-submodule',
+         'action': 'store_true',
+         'help': """
+          Remove the litani submodule and update the definition of LITANI in
+          Makefile-template-defines if the litani submodule is present and
+          the litani command is in PATH.
+          Normally just recommend removal."""},
+    ]
+    args = arguments.create_parser(options, desc).parse_args()
+    arguments.configure_logging(args)
+    return args
+
+################################################################
+
+def validate_cbmc_root(args):
+    args.cbmc_root = args.cbmc_root or Path('.')
+    args.cbmc_root = args.cbmc_root.resolve()
+    if not args.cbmc_root.is_dir():
+        raise UserWarning(f'CBMC root is not a directory: {args.cbmc_root}')
+    if not (args.cbmc_root/util.PROOF_DIR).is_dir():
+        raise UserWarning(f'CBMC root is missing a {util.PROOF_DIR} subdirectory: {args.cbmc_root}')
+    logging.debug('CBMC root: %s', args.cbmc_root)
+    return args
+
+def validate_starter_kit_root(args):
+    starter_kit = repository.starter_kit_root(repo=args.cbmc_root, abspath=True)
+    args.starter_kit_root = args.starter_kit_root or starter_kit
+    if args.starter_kit_root:
+        args.starter_kit_root = args.starter_kit_root.resolve()
+        if not args.starter_kit_root.is_dir():
+            raise UserWarning(f'Starter kit root is not a directory: {args.starter_kit_root}')
+        if not (args.starter_kit_root/util.REPOSITORY_TEMPLATES).is_dir():
```

Comment:
1. I cloned the [FreeRTOS SigV4 project](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/tree/main/test/cbmc)
2. I checked out this branch inside of `test/cbmc/aws-templates-for-cbmc`.
3. Ran ` make develop` and updated my PATH (`export PATH=$PATH:/tmp/cbmc-starter-kit/bin`) such that I can find the binary.
4. I cd'ed to the project root and ran: `cbmc-starter-kit-update --cbmc-root test/cbmc --starter-kit-root test/cbmc/aws-templates-for-cbmc-proofs --no-migrate --no-test-removal --no-update --verbose` and got this error:
```
UserWarning: Starter kit root is missing a template-for-repository subdirectory: /Users/foo/freertos-sigv4/test/cbmc/aws-templates-for-cbmc-proofs
```
Should the path be updated to `args.starter_kit_root/ "src/cbmc_starter_kit" / util.REPOSITORY_TEMPLATES` ?

Reply:
Also, I ran `cbmc-starter-kit-update` from the cbmc root (i.e. `test/cbmc`) and got the same error as above.

Reply:
The starter kit works if I do
```
git clone git@github.com:aws/SigV4-for-AWS-IoT-embedded-sdk.git sigv4
cd sigv4 && git submodule update --init --checkout --recursive
cd sigv4 && cbmc-starter-kit-update --cbmc-root test/cbmc --starter-kit-root test/cbmc/aws-templates-for-cbmc-proofs --no-migrate --no-test-removal --no-update --verbose
```

The starter kit fails as described if I omit the `git submodule update` line.

Reply:
This has been debugged and resulted in a final commit [b9744c3](https://github.com/model-checking/cbmc-starter-kit/pull/95/commits/b9744c38472f63002bf3b97d8e39f7c161c03d57).